UI/org.eclipse.birt.report.debug.ui/src/org/eclipse/birt/report/debug/internal/ui/script/handlers/DebugResourceHandler.java

```diff
@@ -0,0 +1,67 @@
+/*******************************************************************************
+ * Copyright (c) 2008 Actuate Corporation.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ *
+ * Contributors:
+ *  Actuate Corporation  - initial API and implementation
+ *******************************************************************************/
+
+package org.eclipse.birt.report.debug.internal.ui.script.handlers;
+
+import org.eclipse.birt.report.debug.internal.ui.script.launcher.ScriptLaunchShortcut;
+import org.eclipse.core.commands.AbstractHandler;
+import org.eclipse.core.commands.ExecutionEvent;
+import org.eclipse.core.commands.ExecutionException;
+import org.eclipse.core.expressions.IEvaluationContext;
+import org.eclipse.core.resources.IFile;
+import org.eclipse.debug.core.ILaunchConfiguration;
+import org.eclipse.debug.ui.DebugUITools;
+import org.eclipse.jface.viewers.IStructuredSelection;
+import org.eclipse.ui.ISources;
+
+/**
+ * Add the debug report action in the navigate view
+ */
+
+public class DebugResourceHandler extends AbstractHandler {
+
+	protected IFile selectedFile;
+
+	@Override
+	public void setEnabled(Object evaluationContext) {
```

Comment:
still seems excessive to me
we have "visibleWhen" expression that ensures it is a file - why should we calculate this here again and how often?